pkg/daemon/ceph/cleanup/disk.go

```diff
@@ -145,23 +145,42 @@ func (s *DiskSanitizer) returnPVDevice(disk string) []string {
 	return strings.Split(output, ":")
 }
 
-func (s *DiskSanitizer) buildDDArgs(disk string) []string {
-	ddArgs := []string{
-		fmt.Sprintf("if=%s", ddIf),
-		fmt.Sprintf("of=%s", disk),
-		fmt.Sprintf("bs=%s", ddBS),
-		fmt.Sprintf("count=%s", ddCount),
-		fmt.Sprintf("oflag=%s", ddFlags),
+func (s *DiskSanitizer) buildDataSource() string {
+	return fmt.Sprintf("/dev/%s", s.sanitizeDisksSpec.DataSource.String())
+}
+
+func (s *DiskSanitizer) buildShredArgs(disk string) []string {
+	var shredArgs []string
+
+	// If data source is not zero, then let's add zeros at the end of the pass
+	if s.sanitizeDisksSpec.DataSource != cephv1.SanitizeDataSourceZero {
+		shredArgs = append(shredArgs, "--zero")
+	}
+
+	// If this is a quick pass let's just overwrite the first 10MB
+	if s.sanitizeDisksSpec.Method == cephv1.SanitizeMethodQuick {
+		shredArgs = append(shredArgs, fmt.Sprintf("--size=%s", shredBS))
+	}
+
+	// If the data source for randomness is zero
+	if s.sanitizeDisksSpec.DataSource == cephv1.SanitizeDataSourceZero {
```

Comment:
I'm also confused here. :) If the data source is zero, we want to set it to a random source? What is the default if the source is not zero? Then it is actually random? I guess that makes sense.

Reply:
I know it's miss leading. By default `shred` uses a random source, it knows what to do and which random device to use, so I don't want to explicitly build it. Our default is to use zeros, but this means giving to `shred` the full of the path of the random-source which is not so random since it's zeros, but that's the name of the flag :).